src/components/elements/inputs/date/year.js

```diff
@@ -0,0 +1,50 @@
+import cx from 'classnames'
+import Select from 'react-select'
+import 'react-select/dist/react-select.css'
+
+export default ({ className, errors, onCommit, property, value }) => {
+  return (
+    <Select
+      className={cx(className, 'ct-year')}
+      onChange={selected => onCommit(selectYear(selected && selected.value, value))}
+      options={getYears().map(year => {
+        return { label: year, value: year }
+      })}
+      placeholder='Select year'
+      value={getYear(value)} />
+  )
+}
+
+function calculateDay(year, date) {
+  const day = date.getDate()
+  const daysInMonth = new Date(year, date.getMonth() + 1, 0).getDate()
+
+  return day > daysInMonth ? daysInMonth : day
+}
+
+function getYear(value) {
+  return value && value.getFullYear()
+}
+
+function getYears() {
+  const years = []
+  let current = new Date().getFullYear() - 100
+  for (let i = 0; i <= 200; i++) {
```

Comment:
We're doing this in a couple of places -- maybe write a simple range util like [SugarJS](http://sugarjs.com/ranges) has?